web/concrete/elements/files/add_to_sets.php

```diff
@@ -109,10 +109,13 @@ function checkbox($field, $value, $state, $miscFields = array()) {
 
 	if ($_POST['fsNew']) {
 		$type = ($_POST['fsNewShare'] == 1) ? FileSet::TYPE_PUBLIC : FileSet::TYPE_PRIVATE;
-		$fs = FileSet::createAndGetSet($_POST['fsNewText'], $type);
-		//print_r($fs);
-		foreach($files as $f) {
-			$fs->addFileToSet($f);
+		if (preg_match('/[<>;{}?"`]/i', $_POST['fsNewText'])) {
```

Comment:
I just copied that from the single page controller, in the add/create functions I use h() on it.